challengeutils/writeup_attacher.py

```diff
@@ -1,52 +1,154 @@
+"""
+This module is responsible for attaching participant writeup submissions with
+the main challenge queues.  It also archives(copies) projects since there isn't
+currently an elegant way in Synapse to create snapshots of projects.
+"""
+import logging
+import time
 import pandas as pd
+import synapseclient
 from synapseclient.annotations import to_submission_status_annotations
+import synapseutils
 from . import utils
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
 
 
-def append_writeup_to_main_submission(row, syn):
-    '''
-    Helper function that appends the write up synapse id and archived
-    write up synapse id on the main submission
+def _create_archive_writeup(syn, sub):
+    """
+    Creates the archived writeup project
+
+    Args:
+        syn: Synapse object
+        sub: Synapse submission
+
+    Returns:
+        Synapse Project entity
+    """
+    submission_name = sub.entity.name
+    current_time_ms = int(round(time.time() * 1000))
+    archived_name = (f"Archived {submission_name} {current_time_ms} "
+                     f"{sub.id} {sub.entityId}")
+    project_entity = synapseclient.Project(archived_name)
+    entity = syn.store(project_entity)
+    synapseutils.copy(syn, sub.entityId, entity.id)
+    return entity
+
+
+def archive_writeup(syn, submissionid, rearchive=False):
+    """
+    Archive one writeup submission
+
+    Args:
+        syn: Synapse object
+        submissionid: Synapse submission objectId
+        rearchive: Boolean value to rearchive a project or not
+    """
+    # retrieve file into cache and copy it to destination
+    sub = syn.getSubmission(submissionid, downloadFile=False)
+    sub_status = syn.getSubmissionStatus(submissionid)
+    # The .get accounts for if there is no stringAnnos
+    check_if_archived = filter(lambda x: x.get("key") == "archived",
+                               sub_status.annotations.get('stringAnnos', []))
+    archived_entity = list(check_if_archived)
+    # check_if_archived will be an empty list if the annotation doesnt exist
+    if not archived_entity or rearchive:
+        entity = _create_archive_writeup(syn, sub)
+        archived = {"archived": entity.id}
+        sub_status = utils.update_single_submission_status(sub_status,
+                                                           archived)
+        syn.store(sub_status)
+        return entity.id
+    return archived_entity[0]['value']
+
+
+def archive_writeups(syn, evaluation, status="VALIDATED", rearchive=False):
```

Comment:
be more specific with fxn name: `archive_evaluation_writeups` or something

Reply:
What do you think about `archive_writeup_submissions`